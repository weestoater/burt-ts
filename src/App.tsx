import React from "react";
import { BrowserRouter, Routes, Route } from "react-router-dom";

import Header from "./components/general/Header";
import Home from "./pages/Home";
import A11y from "./pages/A11y";
import About from "./pages/About";
import Agile from "./pages/Agile";
import MobRules from "./pages/agile/MobRules";
import { LessonsLearned } from "./pages/agile/LessonsLearned";
import Football from "./pages/Football";
import ReactPage from "./pages/React";

function App() {
  return (
    <BrowserRouter>
      <Header site="burtware.com" />
      <div className="container mt-4">
        <main>
          <Routes>
            <Route path="/" element={<Home />} />
            <Route path="/a11y" element={<A11y />} />
            <Route path="/about" element={<About />} />
            <Route path="/agile" element={<Agile />} />
            <Route path="/mobrules" element={<MobRules />} />
            <Route path="/lessonslearned" element={<LessonsLearned />} />
            <Route path="/football" element={<Football />} />
            <Route path="/react" element={<ReactPage />} />
          </Routes>
        </main>
      </div>
    </BrowserRouter>
  );
}

export default App;
