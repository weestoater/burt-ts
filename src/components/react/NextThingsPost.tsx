import React from "react";

import NextJSLogo from "../../assets/nextjs-logo.png";

export default function NextThingsPost() {
  return (
    <div className="col-md-6 col-sm-6 mb-4">
      <div className="card">
        <div className="card-header">
          <i className="bi bi-binoculars" />
          NextJS things Next...
        </div>
        <div className="card-body">
          <img src={NextJSLogo} alt="NextJS logo" className="circle-right" />
          <p>
            On a recent works 'global hackathon' I did a small PoC using NextJS
            and Bootstrap. I really wasn't sure what to expect with NextJS, but
            somethings were really interesting and others very confusing /
            annoying.
          </p>
          <p>
            The way that NextJs looks after the '<i>server-side</i>'{" "}
            <code>api</code> is very neat - and extensible. I was easily able to
            hook up my data sets and get the PoC showing actual data. This was
            one of the main issues I was aiming to overcome, so such a quick-win
            was really appreciated.
          </p>

          <p>
            Integrating <b>agCharts</b> to show the data was also a breeze,
            because of the support for the ReactJS libraries. Once again, the
            data was being rendered visually really swiftly and I was flying
            through my todo list.
          </p>

          <p>
            Integrating <b>Bootstrap</b> libraries on the other hand was{" "}
            <b>not</b> intuitive / as easy as I had expected it to be. There was
            a <b>lot more googling</b> and <b>stackoverflow</b> searching
            required to find a solution that DID work, rather than the 5 mins it
            takes to implement in a ReactJS project.
          </p>

          <p>
            Overall, I'm not sure how I feel about the NextJS framework, but I
            know several people who swear by it - and they are people I trust
            and respect - so I am probably just being picky. A couple of good
            pluralsight tutorials should probably get me through the learning
            curve and over the top of whatever issues I'm likely to encounter.{" "}
          </p>

          <p>Wish me luck - I'm going to need it.</p>
        </div>
      </div>
    </div>
  );
}
