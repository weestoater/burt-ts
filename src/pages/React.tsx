import React from "react";

import RandomUsersPost from "../components/react/RandomUserPost";
import ReactHooksPost from "../components/react/ReactHooksPost";
import NextThingsPost from "../components/react/NextThingsPost";

export default function ReactPage() {
  return (
    <>
      <div className="row">
        <div className="col-sm-12">
          <h1>React</h1>
        </div>
        <NextThingsPost />
        <ReactHooksPost />
        <RandomUsersPost />
      </div>
    </>
  );
}
