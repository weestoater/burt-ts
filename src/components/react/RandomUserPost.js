import React, {useState, useEffect} from 'react';
import getApi from '../../adapters/commonApi';
import Spinner from '../general/Spinner';
import UserItem from './UserItem';

export default function RandomUsersPost() {

    const [users, setUsers] = useState({});
    const apiUrl = "https://randomuser.me/api/?results=3";

    useEffect(() => {
        getApi(apiUrl).then(setUsers);
      }, []);

    return (
        <div className="col-md-6 col-sm-12 mb-4">
            <div className="card">
                <div className="card-header">
                    <i className="bi bi-filetype-jsx" />
                    Random Users API
                </div>
                <div className="card-body users">
                {users.results ? (
                    users.results.map((user, key) => <UserItem key={key} item={user} /> )
                ) : ( 
                <Spinner /> 
                )}
                </div>
            </div>
        </div>
    );
}
