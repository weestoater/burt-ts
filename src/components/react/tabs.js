import React, { useState, useCallback } from "react";

function Tabs({ children }) {
  const initialTab = children[0].props.label
  const [activeTab, setActiveTab] = useState(initialTab)
  const handleActiveTab = useCallback(label => setActiveTab(label), [])

  const tabs = children.map(child => (
    <button tabIndex={"0"}
      onClick={e => {
        e.preventDefault();
        handleActiveTab(child.props.label);
      }}
      className={
        child.props.label === activeTab
          ? ["tabs__tab", "tabs__tab-active"].join(" ")
          : "tabs__tab"
      }
      key={child.props.label}
    >
      {child.props.tabName}
    </button>
  ))
  
  const tabContent = children.filter(child => child.props.label === activeTab)

  return (
    <div>
      <div className="tabs__box">{tabs}</div>
      <div>{tabContent}</div>
    </div>
  )
}

function Tab(props) {
	return <>{props.children}</>
}

export { Tabs, Tab }