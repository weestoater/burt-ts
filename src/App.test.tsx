import React from "react";
import { render, screen } from "@testing-library/react";
import App from "./App";

test("renders a header link to burtware.com", () => {
  render(<App />);
  const headerText = screen.getByText(/burtware.com/i);
  expect(headerText).toBeInTheDocument();
});

test("renders car videos heading", () => {
  render(<App />);
  const headerText = screen.getByText(/car videos/i);
  expect(headerText).toBeInTheDocument();
});

test("renders music videos heading", () => {
  render(<App />);
  const headerText = screen.getByText(/music videos/i);
  expect(headerText).toBeInTheDocument();
});

test("should match snapshot", () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
