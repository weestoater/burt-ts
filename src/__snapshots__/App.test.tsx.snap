// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg bg-light navbar-light"
  >
    <div
      class="navbar-brand"
    >
      <img
        alt="burt logo"
        class="logo"
        src="assets/burt.png"
      />
      <a
        href="/"
      >
        burtware.com
      </a>
      <a
        class="visually-hidden-focusable"
        href="#main"
      >
        Skip nav
      </a>
    </div>
    <button
      aria-controls="topNavBar"
      aria-expanded="false"
      aria-label="Toggle navigation"
      class="navbar-toggler"
      data-bs-target="#topNavBar"
      data-bs-toggle="collapse"
      data-target="#topNavBar"
      data-toggle="collapse"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="collapse navbar-collapse"
      id="topNavBar"
    >
      <ul
        class="navbar-nav me-auto mb-2 mb-lg-0"
      >
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/about"
          >
            About
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/a11y"
          >
            A11y
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/agile"
          >
            Agile
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/football"
          >
            Football
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/react"
          >
            React
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    class="container mt-4"
  >
    <main
      id="main"
    >
      <div
        class="row mb-3"
      >
        <div
          class="col"
        >
          <h1>
            Welcome to 
            <code>
              burtware
            </code>
          </h1>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-sm-4"
        >
          <h2>
            Thoughts
          </h2>
          <div
            class="card mb-4"
          >
            <div
              class="card-header"
            >
              <i
                class="bi bi-wrench-adjustable-circle"
              />
               Ethos
            </div>
            <div
              class="card-body"
            >
              <p>
                I started this space as an online area for me to experiment with ReactJS and data 'patterns' on AWS. I've not got any desire to add a database or other server-side mechanisms - instead I want to concentrate purely on the front end aspects of rendering
                 
                <code>
                  JSON
                </code>
                 payloads.
                 
              </p>
              <p>
                There isn't any roadmap of what I will build, but I am iterating around the football details, using the Motherwell F.C. results as my data source. I am 'hand-rolling' the data into two
                 
                <code>
                  .json
                </code>
                 files: mfc-goals & mfc-matches, then creating a number of components to render the results onto the page.
              </p>
            </div>
          </div>
          <div
            class="card mb-4"
          >
            <div
              class="card-header"
            >
              <i
                class="bi bi-bug"
              />
               trial & error
            </div>
            <div
              class="card-body"
            >
              <p>
                Ideally I want to learn as much as possible in building this site, as I have the freedom to make as many mistakes as I like.
              </p>
              <p>
                It's only a 
                <code>
                  git pull
                </code>
                 away from being removed or over-written, so I don't want to be precious.
              </p>
            </div>
          </div>
        </div>
        <div
          class="col-sm-4"
        >
          <h2>
            Car Videos
          </h2>
          <figure>
            <iframe
              src="https://www.youtube.com/embed/i0Z1VKWf3RM"
              title="YouTube video player 1"
            />
            <figcaption>
              Range Rover Sport vs spillway
            </figcaption>
          </figure>
          <figure>
            <iframe
              src="https://www.youtube.com/embed/R6ygiL366TE"
              title="YouTube video player 2"
            />
            <figcaption>
              Guy Martin in highly modified Land Rover
            </figcaption>
          </figure>
        </div>
        <div
          class="col-sm-4"
        >
          <h2>
            Music Videos
          </h2>
          <figure>
            <iframe
              src="https://www.youtube.com/embed/iTKJ_itifQg"
              title="YouTube video player 3"
            />
            <figcaption>
              depeche Mode - I Feel You
            </figcaption>
          </figure>
          <figure>
            <iframe
              src="https://www.youtube.com/embed/xy37XdAwH_A"
              title="YouTube video player 4"
            />
            <figcaption>
              Lindemann - Home Sweet Home (live)
            </figcaption>
          </figure>
        </div>
      </div>
    </main>
  </div>
</div>
`;
